lib/server.js

```diff
@@ -1,29 +1,45 @@
-let { loadExtension, repr } = require("./util");
+let { loadExtension, abort, repr } = require("./util");
 
-exports.static = function(config, webroot) {
+let DEFAULTS = {
+	host: "localhost",
+	port: 3000
+};
+
+exports.static = (config, webroot) => {
 	let donny = loadExtension("donny", "failed to activate server");
-	let [host, port] = parse(config);
+	let [host, port] = parseHost(config);
 
 	donny({ port, bind: host, webroot }).
 		then(() => {
 			console.error(`serving ${repr(webroot)} at http://${host}:${port}`);
 		});
 };
 
-exports.live = function(config, root) {
+exports.live = (config, root) => {
 	let liveServer = loadExtension("live-server", "failed to activate live-server");
-	let [host, port] = parse(config);
+	let [host, port] = parseHost(config);
 
 	liveServer.start({ port, host, root, open: false });
 };
 
-function parse(config) {
-	let i = config.lastIndexOf(":");
-	if(i === -1) { // port only
-		return ["0.0.0.0", config];
-	}
+exports._parseHost = parseHost;
 
-	let host = config.substr(0, i);
-	let port = config.substr(i + 1);
-	return [host, port];
+function parseHost(config) {
```

Comment:
So... parseHost returns the port and the host? That naming seems weird to me.

Reply:
Yeah, ditto; it's a perennial issue*. I had considered returning `[hostname, port]` instead to highlight the distinction between narrow and broad interpretation of "host", but that seemed fraught as well (think IPs).

The term "config" struck me as misleading (indeed, I might change the signature to `(value)`), which is why I changed the name in the first place. Rock, hard place.

* Pretty sure we'd discussed this elsewhere before. 🤔

Reply:
Ok 👌